### PR TITLE
Fix component catalog UI sorting

### LINKED ISSUE #2297: Catalog connectors are not properly sorted by name in the 'components catalog' panel
**Describe the issue**

In the 'component catalogs' panel the entries are not sorted alphabetically if one or more BYO component catalogs is installed:

image

**To Reproduce**
1. Install Elyra/master
2. Install one or more of the example catalogs (see [these instructions](https://github.com/ptitzler/examples/tree/example-components-catalog/component-catalog-connectors))
3. Open the 'component catalogs' panel 
4. Click `+` to add the catalog(s)

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: 3.3/master

### PR DESCRIPTION
**Pull request by karlaspuldaro**
Fixes #2297

### What changes were proposed in this pull request?
Sort metadata schema by title, which is what is used by the UI for display, instead of name

### How was this pull request tested?
Manual testing. Added KFP component catalog connector from [examples](https://github.com/ptitzler/examples/tree/example-components-catalog/component-catalog-connectors/kfp-example-components-connector). 
In the component catalog widget, click the + to add new and check the list is now sorted:
image

Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Review by kiersten-stokes — APPROVED** [review]
LGTM!

Screen Shot 2021-11-30 at 5 08 08 PM

**Comment by ptitzler** [comment]
Could we sort case insensitive?

File: packages/metadata-common/src/AddMetadataButton.tsx [hunk 1]
```diff
@@ -58,15 +58,9 @@ export const AddMetadataButton = (
...
-    if (a.name > b.name) {
-      return 1;
-    } else if (a.name < b.name) {
-      return -1;
-    } else {
-      return 0;
-    }
-  });
+  const sortedSchema = props.schemas?.sort((a, b) =>
+    a.title.localeCompare(b.title)
```
Comment (karlaspuldaro) [inline packages/metadata-common/src/AddMetadataButton.tsx:62]: Using [localeCompare](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#sort_an_array) to enable case-insensitive sorting.

**Review by marthacryan — APPROVED** [review]

**Comment by marthacryan** [comment]
Haven't tried this locally but the code changes look good to me

**Review by ajbozarth — APPROVED** [review]
Tested locally

LGTM

**Pull request merged**